srv/salt/_modules/osd.py

```diff
@@ -174,11 +174,25 @@ def rescinded():
     return _ids
 
 
+def orphaned():
+    """
+    Return array of CRUSH hosts that are orphaned.
+    """
+    osd_tree = tree_from_master()
+    #orphaned_hosts = list(filter(lambda x: x['type'] == 'host' and len(x['children']) == 0, osd_tree['nodes']))
+    #orphaned_host_names = list(map(lambda x: x['name'], orphaned_hosts))
+    orphaned_hosts = [x for x in osd_tree['nodes']
+                        if x['type'] == 'host' and len(x['children']) == 0]
+    orphaned_host_names = [x['name'] for x in orphaned_hosts]
+    log.debug("Orphaned CRUSH hosts: {}".format(','.join(orphaned_host_names)))
```

Comment:
I cannot find any occurence of 'Orphaned CRUSH hosts' in the logs

Reply:
osd.orphaned runs on the master minion - are you looking in the right log? I can see this line in the latest run http://137.74.25.20:8081/ubuntu-2018-08-21_22:01:10-suse:tier1:stage-5-ses6---basic-openstack/